### ISSUE TITLE
Remove the gh-pages branch and enforce HTTPS

### ISSUE THREAD

**Issue by Potherca** [ylebre/spectral#7]
Currently this repository has two branches, `gh-pages` and `master`.

Recently,  Github has made [the source branch for Github Pages configurable]. Thus a `gh-pages` branch is no longer needed.

To avoid confusion and to keep things easier to maintain I would strongly suggest removing the `gh-pages` branch and  configuring the master branch for publishing.

![](https://help.github.com/assets/images/help/pages/select-gh-pages-or-master-as-source.png)

Github Pages also allows for [https-only mode] which I would also recommend enabling (after #5 has been merged).

![](https://help.github.com/assets/images/help/pages/enforce-https-checkbox.png)
 
[https-only mode]: https://help.github.com/articles/securing-your-github-pages-site-with-https/
[the source branch for Github Pages configurable]: https://help.github.com/articles/configuring-a-publishing-source-for-github-pages/

**Comment by ylebre** [comment]
both fixed, gh-pages branch is now removed.

**Comment by Potherca** [comment]
The change in branch seems to have caused a new error:

---
> https://raw.githubusercontent.com/ylebre/spectral/gh-pages/data.json 
>
> Failed to load resource: the server responded with a status of 404 (Not Found)
---
> simply-edit.js:202 No data found, starting with empty dataset
---

It looks like this is caused by hard-coded values `gh-pages` for `repoBranch` in [simply-edit.js](https://cdn.simplyedit.io/1/simply-edit.js):

```js
{
    github: {
        repoName: null,
        repoUser: null,
        repoBranch: "gh-pages",
        dataFile: "data.json",
        getRepoInfo: function(endpoint) {
            var result = {};
            var parser = document.createElement('a');
            parser.href = endpoint;
            var pathInfo;
            pathInfo = parser.pathname.split("/");
            if (parser.pathname.indexOf("/") === 0) {
                pathInfo.shift();
            }
            if (parser.hostname == "github.com") {
                result.repoUser = pathInfo.shift();
                result.repoName = pathInfo.shift();
                result.repoBranch = "master";
            } else {
                result.repoUser = parser.hostname.split(".")[0];
                result.repoName = pathInfo.shift();
                result.repoBranch = "gh-pages";
            }
            var repoPath = pathInfo.join("/");
            repoPath = repoPath.replace(/\/$/, '');
            result.repoPath = repoPath;
            return result;
        }
    }
}
```

Should I create a new issue for this?
